fdbclient/SpecialKeySpace.actor.cpp

```diff
@@ -574,16 +574,16 @@ ACTOR Future<Void> commitActor(SpecialKeySpace* sks, ReadYourWritesTransaction*
 	state RangeMap<Key, std::pair<bool, Optional<Value>>, KeyRangeRef>::Ranges ranges =
 	    ryw->getSpecialKeySpaceWriteMap().containedRanges(specialKeys);
 	state RangeMap<Key, std::pair<bool, Optional<Value>>, KeyRangeRef>::iterator iter = ranges.begin();
-	state std::set<SpecialKeyRangeRWImpl*> writeModulePtrs;
+	state std::vector<SpecialKeyRangeRWImpl*> writeModulePtrs;
 	while (iter != ranges.end()) {
 		std::pair<bool, Optional<Value>> entry = iter->value();
 		if (entry.first) {
 			auto modulePtr = sks->getRWImpls().rangeContaining(iter->begin())->value();
-			writeModulePtrs.insert(modulePtr);
+			writeModulePtrs.push_back(modulePtr);
```

Comment:
Thanks for fixing this.
This `modulePtr` can be duplicate, that's the reason why I used `std::set`.
Since the size of the vector is always small, I guess we can check if it exists before inserting it every time?
Or using a set as reference but using a vector to traverse?

Reply:
Thanks - I deduplicate with a set now